### DISCUSSION TITLE
How to solve"Model is infeasible or unbounded Best objective -, best bound -, gap -"

### DISCUSSION THREAD

**Discussion by wendy444444** [yalmip/YALMIP#1115] [General discussion]
Dear Johan，I read your other answers, so I try to solve it. Some constraints are modified and explicit boundaries are added, and solved the original problem. But there are still some problems unsolved. I tried to use gurobi_iis to find the formula that went wrong, but I have no clue. Please help me again. This is the result of operation.
```
Set parameter Username
Set parameter IntFeasTol to value 1e-06
Set parameter ScaleFlag to value 1
Set parameter BarIterLimit to value 2000000000
Set parameter CrossoverBasis to value 0
Set parameter MinRelNodes to value 0
Set parameter NodeMethod to value 1
Set parameter PumpPasses to value 0
Set parameter ZeroObjNodes to value 0
Set parameter AggFill to value 10
Set parameter PreSparsify to value 0
Set parameter TuneTrials to value 2
Academic license - for non-commercial use only - expires 2022-06-11
Gurobi Optimizer version 9.5.1 build v9.5.1rc2 (win64)
Thread count: 2 physical cores, 4 logical processors, using up to 4 threads
Optimize a model with 6940 rows, 1140 columns and 13591 nonzeros
Model fingerprint: 0x49adfbe9
Variable types: 888 continuous, 252 integer (252 binary)
Coefficient statistics:
  Matrix range     [8e-02, 3e+03]
  Objective range  [1e+00, 1e+00]
  Bounds range     [1e+00, 1e+00]
  RHS range        [1e+00, 3e+03]
Presolve time: 0.02s

Explored 0 nodes (0 simplex iterations) in 0.09 seconds (0.00 work units)
Thread count was 1 (of 4 available processors)

Solution count 0

Model is infeasible or unbounded
Best objective -, best bound -, gap -
时间已过 16.140629 秒。
Set parameter Username
Academic license - for non-commercial use only - expires 2022-06-11

IIS computed: 1 constraints, 0 bounds
IIS runtime: 0.11 seconds (0.00 work units)
```

**Comment by wendy444444** [comment]


**Reply by johanlofberg** [reply]
https://yalmip.github.io/debugginginfeasible

**Reply by wendy444444** [reply]
Thank you for your support,I've already followed the steps mentioned in the link.But it still has no solution. What should I do,please?
```
Set parameter Username
Set parameter IntFeasTol to value 1e-06
Set parameter ScaleFlag to value 1
Set parameter BarIterLimit to value 2000000000
Set parameter CrossoverBasis to value 0
Set parameter MinRelNodes to value 0
Set parameter NodeMethod to value 1
Set parameter PumpPasses to value 0
Set parameter ZeroObjNodes to value 0
Set parameter AggFill to value 10
Set parameter PreSparsify to value 0
Set parameter TuneTrials to value 2
Academic license - for non-commercial use only - expires 2022-06-11
Gurobi Optimizer version 9.5.1 build v9.5.1rc2 (win64)
Thread count: 2 physical cores, 4 logical processors, using up to 4 threads
Optimize a model with 6979 rows, 1184 columns and 13679 nonzeros
Model fingerprint: 0x11183c6d
Variable types: 888 continuous, 296 integer (296 binary)
Coefficient statistics:
  Matrix range     [8e-02, 3e+03]
  Objective range  [1e+00, 1e+00]
  Bounds range     [1e+00, 1e+00]
  RHS range        [8e-01, 3e+03]
Found heuristic solution: objective 0.0000000
Presolve removed 6979 rows and 1184 columns
Presolve time: 0.02s
Presolve: All rows and columns removed

Explored 0 nodes (0 simplex iterations) in 0.06 seconds (0.00 work units)
Thread count was 1 (of 4 available processors)

Solution count 1: 0 

Optimal solution found (tolerance 1.00e-04)
Best objective 0.000000000000e+00, best bound 0.000000000000e+00, gap 0.0000%
```

**Reply by johanlofberg** [reply]
I have no idea what you talk about now, because you now show a case of a trivially feasible problem. The problem you sent before was infeasible, but it showed no indication that you had tried to eliminate stuff that did not cause infeasibility (and it didn't even show that you had removed the objective to find out if it was infeasible or unbounded)

**Reply by wendy444444** [reply]
After reading your link, I tried to use 2, 5 and 7 of them, and then the program worked. However, I am sorry that I cannot paste the program now, so I would like to ask you how to deal with the problem

**Reply by johanlofberg** [reply]
You cannot just a remove bunch of constraints and then conclude that it runs and stop there. Once you figured out some sets of constraints which cause infeasibility when added on a maximally reduced model, it is up to you to figure out why they cause infeasibility.  if it is reasonable that they cause infeasibility, if it is a bug in your implementation which cause it or a bug in your theoretical model, or if you are solving an instance which just is infeasible due to the data being used.

**Reply by wendy444444** [reply]
I didn't just delete that constraint, I found it conflicted with the other constraints, so I changed it, and it became feasible, so I thought it would work. But I don't understand the meaning of maximally reduced model. Could you please explain it to me

**Reply by johanlofberg** [reply]
When you search for what causes infeasibility, you remove everything which doesn't cause infeasibility, i.e. you simplify and reduce the model as much as absolutely possible, before trying to figure out why and how the infeasibility occurs. No reason to have more stuff in the model.

**Reply by wendy444444** [reply]
Thank you for your kindness.I'm not sure but I remember replacing 
`C = [C,0.9*0.9*UB*UB<=V<=1.1*1.1*UB*UB];` 
 with `C = [C,0.9*0.9<=V<=1.1*1.1];` and  then  it worked. As for what you explained to me, I also want to ask that there is a correlation between my constraints, so what should I do?

**Reply by johanlofberg** [reply]
I don't know what you are asking or claiming.

**Reply by wendy444444** [reply]
What I mean is that part of my constraints are complicated, and deleting one may require changing others or deleting others

**Reply by johanlofberg** [reply]
Well, sure. You can have a model with N constrints which is infeasible, but removing any one of them renders the model feasible. It is your job to understand your model and analyze the consequences of changing things to understand what causes infeasibility. No one can do that for you

**Reply by wendy444444** [reply]
Thank you for your answer. How did you determine that my program is not feasible? The result I obtained using `sol = optimize(C,Objective);if sol.problem==0;display('Feasible');else;display('Infeasible');end` is feasible.

**Reply by johanlofberg** [reply]
Gurobi said it was infeasible or unbounded. Hence remove the objective to check if it still complains (thus about feasibility) or not (i.e. it was unbounded)

**Reply by wendy444444** [reply]
Thank you for your wise advice, the program is running, but it took a long time, these page appears, what should I do, please?
![mmexport1651833794562](https://user-images.githubusercontent.com/103562806/167117422-27065950-d104-4bc4-a6f0-b8bfe1928613.jpg)
![mmexport1651833845141](https://user-images.githubusercontent.com/103562806/167117427-31b34907-30e0-4218-a82f-7597eb3e303e.jpg)

**Reply by johanlofberg** [reply]
What do you mean with what to do. The solver is working, now you just have to wait until in completes, which might take a second/ hour/day/month/year/eons https://yalmip.github.io/slowmilp

**Reply by wendy444444** [reply]
Thank you very much. I will try to do it.
